File: AtCoder/ABC/C/228/C.rb
n, k = gets.chomp.split.map(&:to_i)
p = Array.new
n.times do |i| 
    p << [-gets.chomp.split.map(&:to_i).sum, i]
end

p.sort!

clear_score = p[k - 1][0].abs - 300
pass = Array.new(n, false)
p.each do |val, i|
    pass[i] = true if val.abs >= clear_score
end

for i in 0...n do
    puts pass[i] ? "Yes" : "No"
end

File: AtCoder/ABC/C/228/C.py
def main():
    n, k = map(int, input().split())
    p = list()
    for i in range(n):
        p.append([-sum(list(map(int, input().split()))), i])
    p.sort()

    clear_score = abs(p[k - 1][0]) - 300
    judge = [False] * n
    
    for i in range(n): if abs(p[i][0]) >= clear_score: judge[p[i][1]] = True
    for i in range(n): print("Yes" if judge[i] else "No")


if __name__ == '__main__':
    main()